src/magic_folder/error.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+    unicode_literals,
+)
+
+import attr
+
+
+@attr.s
+class PublicError(Exception):
```

Comment:
As discussed above, I don't think we need this exception type.

Reply:
The reason I added this is to be extra-explicit about errors that can be shown to users (and so that when writing / handling them, there's an extra reminder that you're writing for an end-user and not a developer or other expert).